src/status_printer.h

```diff
@@ -100,7 +100,10 @@ struct StatusPrinter : Status {
   Explanations* explanations_ = nullptr;
 
   /// The custom progress status format to use.
-  const char* progress_status_format_;
+  const char* progress_status_format_ = nullptr;
+
+  /// The progress status format to use by default
+  static constexpr const char* kDefaultProgressStatusFormat = "[%f/%t] ";
```

Comment:
nit: `static constexpr` members need a definition in the corresponding .cc file or some compilers will complain that it is missing (especially when compiling with `-O0`). In this case you can simply add `constexpr const char* StatusPrinter::kDefaultProgressStatusFormat;` (with no value) to `status_printer.cc`.

However, we do not need to expose the value publicly, so for simplicity, just define it as a const variable in the constructor that uses it.

Also `const char kName[] = "..."` is preferable to `static const char* kName = "..."` because the latter introduces an unnecessary relocation in the generated code / data segment.

```
StatusPrinter::StatusPrinter(...) {
    ...
   const char kDefaultProgressStatusFormat[] = "[%f/%t] ";
   ...
}
```

Sorry for the nitpicks, this is a great PR :-)

Reply:
It's no problem at all, this is great feedback. :)